test/uk/gov/hmrc/leakdetection/scanner/MatchedResultSpec.scala

```diff
@@ -0,0 +1,134 @@
+          val strippedElipses = truncated.lineText.stripPrefix("[…] ").stripSuffix(" […]").replaceAll(" \\[…\\] ", "")
+          strippedElipses.length should be <= limit
+      }
+    }
+
+    "contain all matches if their total length is <= limit" in {
+      forAll(genMatchedResult, Gen.posNum[Int], minSuccessful(500)) {
+        case (initialResult, limit) =>
+          val initialTotalLengthOfAllMatches =
+            initialResult.matches.map(m => initialResult.lineText.substring(m.start, m.end).length).sum
+          if (initialTotalLengthOfAllMatches <= limit) {
+            val truncated = truncate(initialResult, limit)
+            initialResult.matches.foreach { m =>
+              val leakValue = initialResult.lineText.substring(m.start, m.end)
+              assert(truncated.lineText.contains(leakValue))
+            }
+          }
+      }
+    }
+
+    "don't put […] inbetween consecutive matches" in {
+      val initialResult = genMatchedResult.sample.get.copy(
+        lineText = "abc XXFF xyz",
+        matches  = List(Match(4, 6), Match(6, 8))
+      )
+
+      truncate(initialResult, 4).lineText shouldBe "[…] XXFF […]"
+    }
+
+    "contain as many matches as still below limit" in {
+
+      val initialResult = genMatchedResult.sample.get.copy(
+        lineText = "abc AA def BB ghi CC xyz",
+        matches = {
+          val matchedAA = Match(4, 6)
+          val matchedBB = Match(11, 13)
+          val matchedCC = Match(18, 19)
+          List(matchedAA, matchedBB, matchedCC)
+        }
+      )
+
+      val limit = "AABB".length
+
+      truncate(initialResult, limit).lineText shouldBe "[…] AA […] BB […]"
+    }
+
+    "be idempotent" in {
+      forAll(genMatchedResult, Gen.posNum[Int], minSuccessful(500)) {
+        case (initialResult, limit) =>
+          val truncatedOnce  = truncate(initialResult, limit)
+          val truncatedAgain = truncate(truncatedOnce, limit)
+
+          truncatedOnce shouldBe truncatedAgain
+      }
+    }
+
+    "include a flag if text was truncated (to show info in the UI)" in {
+      forAll(genMatchedResult, Gen.posNum[Int], minSuccessful(500)) {
+        case (initialResult, limit) =>
+          val res = truncate(initialResult, limit)
+          if (initialResult.lineText.length > limit) {
+            assert(res.isTruncated)
+          }
+      }
+    }
+  }
+
+  val genMatchedResult: Gen[MatchedResult] =
+    for {
+      lineText <- Gen.alphaStr.map(_ + "x")
+      matches  <- genConsecutiveMatches(lineText)
+    } yield
+      MatchedResult(
+        scope       = "scope",
+        lineText    = lineText,
+        lineNumber  = 1,
+        ruleId      = "ruleId",
+        description = "description",
+        matches     = matches)
+
+  def genMatch(lineText: String): Gen[Match] =
+    (for {
+      start <- Gen.choose(0, lineText.length)
+      end   <- Gen.choose(start, lineText.length)
+    } yield {
+      Match(start, end)
+    }).retryUntil(m => m.start != m.end)
```

Comment:
Alternatively: 

```scala
  def genMatch(lineText: String): Gen[Match] =
    for {
      start <- Gen.choose(0, lineText.length)
      end   <- Gen.choose(start, lineText.length)
      if start != end
    } yield Match(start, end)
```

I understand that you get 100k retries with retryUntil, and you only get as many as scalatest's maxDiscards setting, so maybe this isn't exactly what you're after.

Reply:
Yes, I more likely to not fail with ~100k~ 10k retries that `retryUntil` provides by default

Reply:
I forgot the golden 'check before you comment' rule.  10k it is :)